### PR TITLE
refactor libraries based on conditional directive changes

### PR DESCRIPTION
**Pull request by devoncarew**
- refactor libraries based on conditional directive changes
- remove `lib/usage_io.dart`
- remove `lib/usage_html.dart`
- update the readme and samples for API changes
- fix https://github.com/dart-lang/usage/issues/66

@collinjackson

### CODE REVIEW

File: lib/src/usage_impl_html.dart [hunk 1]
```diff
@@ -25,6 +25,26 @@ Future<Analytics> createAnalytics(
   ));
 }
 
+class AnalyticsHtml extends AnalyticsImpl {
```
Comment (collinjackson) [inline lib/src/usage_impl_html.dart]: I'm not sure this class needs to exist, we could follow the same pattern as usage_impl_flutter.dart and allocate an AnalyticsImpl in createAnalytics

File: lib/src/usage_impl_io.dart [hunk 2]
```diff
@@ -27,6 +27,19 @@ Future<Analytics> createAnalytics(
   ));
 }
 
+class AnalyticsIO extends AnalyticsImpl {
```
Comment (collinjackson) [inline lib/src/usage_impl_io.dart]: ditto

**Comment by collinjackson** [comment]
lgtm, I left had minor refactoring suggestion but you can take it or leave it

File: lib/src/usage_impl_html.dart (hunk 1 above)
Comment (devoncarew) [inline lib/src/usage_impl_html.dart]: I have reached in and used specific impl classes from usage clients before (`AnalyticsHtml`, ...) but am no longer sure that that is necessary. It's in a package private library `lib/src/`, so we can refactor to remove it later w/o changing the package version.

**Comment by devoncarew** [comment]
Thanks! Will keep those classes for now, but we can remove later w/o breaking api compatibility.

**Pull request merged**